src/parsers.rs

```diff
@@ -1,13 +1,276 @@
 //! This module implements Temporal Date/Time parsing functionality.
 
+use crate::{
+    options::{DisplayCalendar, DisplayOffset, DisplayTimeZone},
+    Sign, TemporalError, TemporalResult, TemporalUnwrap,
+};
 use alloc::format;
-
-use crate::{TemporalError, TemporalResult, TemporalUnwrap};
-
 use ixdtf::parsers::{
     records::{Annotation, DateRecord, IxdtfParseRecord, TimeRecord, UtcOffsetRecordOrZ},
     IxdtfParser,
 };
+use writeable::{impl_display_with_writeable, Writeable};
+
+// TODO: Move `Writeable` functionality to `ixdtf` crate
+
+#[derive(Debug, Clone, Copy, PartialEq, Eq, PartialOrd, Ord)]
+pub enum Precision {
+    Auto,
+    Minute,
+    Digit(u8),
+}
+
+pub struct FormattableTime {
+    pub hour: u8,
+    pub minute: u8,
+    pub second: u8,
+    pub nanosecond: u32,
+    pub precision: Precision,
+    pub include_sep: bool,
+}
+
+impl Writeable for FormattableTime {
+    fn write_to<W: core::fmt::Write + ?Sized>(&self, sink: &mut W) -> core::fmt::Result {
+        write_padded_u8(self.hour, sink)?;
+        if self.include_sep {
+            sink.write_char(':')?;
+        }
+        write_padded_u8(self.minute, sink)?;
+        if self.precision == Precision::Minute {
+            return Ok(());
+        }
+        if self.include_sep {
+            sink.write_char(':')?;
+        }
+        write_padded_u8(self.second, sink)?;
+        if self.nanosecond == 0 {
+            return Ok(());
+        }
+        sink.write_char('.')?;
+        write_nanosecond(self.nanosecond, self.precision, sink)?;
+
+        Ok(())
+    }
```

Comment:
Suggestion: your `impl Writeable` may benefit from a `write_len` function since it looks fairly cheap to pre-compute. You could add one later.